src/tests/ftest/server/multiengine_persocket.py

```diff
@@ -196,28 +217,32 @@ def test_multiengines_per_socket(self):
         :avocado: tags=server
         :avocado: tags=test_multiengines_per_socket
         """
-        # (1) Storage prepare --scm-ns-per-socket
+        # (1) Scm reset and prepare --scm-ns-per-socket
         step = 1
-        self.log.info("===(%s)===Storage prepare --scm-ns-per-socket", step)
-        self.daos_server_storage_prepare_reset(step)
+        self.log.info("===(%s)===Scm reset and prepare --scm-ns-per-socket", step)
         engines_per_socket = self.params.get(
             "engines_per_socket", "/run/server_config/*", default=1)
-        self.daos_server_storage_prepare_ns(step, engines_per_socket)
+        self.daos_server_scm_reset(step)
+        self.host_reboot(self.hostlist_servers)
+        self.daos_server_scm_prepare_ns(1.1, engines_per_socket)
+        self.host_reboot(self.hostlist_servers)
+        self.daos_server_scm_prepare_ns(1.2, engines_per_socket)
+
         cmd = "/usr/bin/ls -l /dev/pmem*"
-        results = run_pcmd(self.hostlist_servers, cmd, timeout=90)
+        results = run_remote(self.log, self.hostlist_servers, cmd, timeout=90)
```

Comment:
Replace lines 231-246 with:
```
    if not self.wait_for_result(self, self.check_pmem, 100, hosts=hosts, count=4, cmd_timeout=10, verbose=True):
            self.fail("4 pmem devices not found on all hosts within 100 seconds.")
```

Which will require the addition of this method:
```
    def check_pmem(self, hosts, count, cmd_timeout=10, verbose=True):

        storage = StorageInfo(self.log, hosts)
        try:
            storage.scan()            
        except StorageException:
            self.log.debug("Problem running StorageInfo.scan()"
        return (len(storage.pmem_devices) == count)
```